python/ray/data/_internal/planner/sort.py

```diff
@@ -31,11 +33,23 @@ def fn(
         ctx: TaskContext,
     ) -> Tuple[List[RefBundle], StatsDict]:
         blocks = []
+        block_schemas: List[Optional[Union[type, pa.lib.Schema]]] = []
+        valid_block_schema: Optional[Union[type, pa.lib.Schema]] = None
         for ref_bundle in refs:
-            for block, _ in ref_bundle.blocks:
+            for block, block_metadata in ref_bundle.blocks:
                 blocks.append(block)
+                if block_metadata.schema is not None and valid_block_schema is None:
+                    valid_block_schema = block_metadata.schema
+                block_schemas.append(block_metadata.schema)
         if len(blocks) == 0:
             return (blocks, {})
+        if isinstance(valid_block_schema, pa.lib.Schema):
+            unified_schema = unify_schemas(block_schemas)
+        else:  # Covers both simple-type and None cases.
+            if isinstance(valid_block_schema, type):
+                assert all([b == valid_block_schema for b in block_schemas])
+            unified_schema = valid_block_schema
+        _validate_key_fn(unified_schema, key)
```

Comment:
could we refactor https://github.com/ray-project/ray/blob/master/python/ray/data/_internal/plan.py#L427-L453 into a common method, and reuse the same code here?